File: src/test/java/se/johannesdahlgren/aoc23/Day4Test.java
package se.johannesdahlgren.aoc23;

import static org.assertj.core.api.Assertions.assertThat;

import org.junit.jupiter.api.Disabled;
import org.junit.jupiter.api.Test;

class Day4Test {

  @Test
  void part1Example() {
    Day4 day4 = new Day4("day4example.txt");
    int points = day4.pointsOfWinningNumbers();
    assertThat(points).isEqualTo(13);
  }

  @Test
  void part1() {
    Day4 day4 = new Day4("day4.txt");
    int points = day4.pointsOfWinningNumbers();
    assertThat(points).isEqualTo(23235);
  }


  @Test
  void part2Example() {
    Day4 day4 = new Day4("day4example.txt");
    int numberOfScratchCards = day4.numberOfScratchCards();
    assertThat(numberOfScratchCards).isEqualTo(30);
  }

  @Disabled("Slow, around 10 seconds")
  @Test
  void part2() {
    Day4 day4 = new Day4("day4.txt");
    int numberOfScratchCards = day4.numberOfScratchCards();
    assertThat(numberOfScratchCards).isEqualTo(5920640);
  }
}

File: src/test/kotlin/se/johannesdahlgren/aoc22/Day4Test.kt
package se.johannesdahlgren.aoc22

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class Day4Test {

    @Test
    fun part1Example() {
        val day4 = Day4("day4Example")
        assertThat(day4.numberOfAssignmentPairsFullyContained()).isEqualTo(2)
    }

    @Test
    fun part1() {
        val day4 = Day4("day4")
        assertThat(day4.numberOfAssignmentPairsFullyContained()).isEqualTo(515)
    }

    @Test
    fun part2Example() {
        val day4 = Day4("day4Example")
        assertThat(day4.numberOfAssignmentPairsOverlap()).isEqualTo(4)
    }

    @Test
    fun part2() {
        val day4 = Day4("day4")
        assertThat(day4.numberOfAssignmentPairsOverlap()).isEqualTo(883)
    }
}